vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java

```diff
@@ -352,6 +369,50 @@ private void setupSearchFieldForExperimentTabs(String experimentName,
         }
       });
     }
+
+    private Result<?, ?> registerBatchAndSamples(BatchRegistrationContent batchRegistrationContent,
+        List<SampleRegistrationContent> sampleRegistrationContent) {
+      return registerBatchInformation(batchRegistrationContent).onValue(
+          batchId -> registerSamples(batchId, sampleRegistrationContent));
+    }
+
+    private Result<BatchId, ResponseCode> registerBatchInformation(
+        BatchRegistrationContent batchRegistrationContent) {
+      return batchRegistrationService.registerBatch(batchRegistrationContent.batchLabel(),
+          batchRegistrationContent.isPilot()).onError(responseCode -> displayRegistrationFailure());
+    }
+
+    private void registerSamples(BatchId batchId,
+        List<SampleRegistrationContent> sampleRegistrationContents) {
+      List<SampleRegistrationRequest> sampleRegistrationRequests = sampleRegistrationContents.stream()
+          .map(sampleRegistrationContent -> createSampleRegistrationRequest(batchId,
+              sampleRegistrationContent)).toList();
+      sampleRegistrationService.registerSamples(sampleRegistrationRequests, projectId)
+          .onError(responseCode -> displayRegistrationFailure());
+    }
```

Comment:
Way better, great improvement. Now, I still have a concern, you method does to things: the creation of registration requests and the actual registration. 

Since a function should do only one thing, I would argue to extract the conversion to registration requests and pass the requests to this function, which does then only the registration.